serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc

```diff
@@ -1 +1,284 @@
+
+Given this use case, what happens behind the scenes is a sequence of steps that should be executed to complete the order, as shown in the picture:
+
+.Example Order fulfillment
+image::use-cases/orchestration-based-saga-pattern/order-fulfillment-example.png[]
+
+In the example, there are the mainly following steps:
+
+<1> _Order checkout_: the user confirms the order with the chosen items, payment method, and delivery address.
+<2> _Stock reservation_: where this item from the order is selected from the stock and reserved, being removed from the total stock item.
+<3> _Payment processing_: based on the payment method and the user information, for instance, the credit card, the credit is checked and allocated to that payment.
+<4> _Shipping processing_: it represents a mechanism that could communicate with third-party systems responsible to deliver orders or even an internal system responsible for logistics and delivery.
+
+The below picture makes it possible to visualize the two main flows in case of success or failure during the order fulfillment flow, which describes a Saga pattern.
+
+.Example Order fulfillment Saga
+image::use-cases/orchestration-based-saga-pattern/order-fulfillment-saga.png[]
+
+In the first case, the happy path, the item stock is reserved. After that, the payment is successfully processed, the order shipping is scheduled, and the flow is completed for that order.
+
+The second case describes an example of a failure or any error thrown during the shipping process for a given order. In this situation, a sequence of compensation actions is executed, which cancels the payment, and releases the stock for the items related to that order.
+
+== Saga orchestration with Serverless workflow
+
+As a developer, you can use link:{spec_doc_url}[Serverless Workflow] to create a Saga defining the participants' calls with all the steps and compensations.
+
+In this scenario, Kogito Serverless Workflow plays the Saga Executor Coordinator (SEC) role, which orchestrates the calls to the participants in the Saga.
+
+The order fulfillment example showcases how to define the Saga with Kogito Serverless Workflow. The source code can be found here in link:{kogito_sw_examples_url}/serverless-workflow-saga-quarkus[serverless-workflow-saga-quarkus].
+
+The workflow definition used to define the order saga is link:{kogito_sw_examples_url}/serverless-workflow-saga-quarkus/src/main/resources/order-saga-error-handling.sw.json[order-saga-error-handling.sw.json].
+
+.Example Order fulfillment Saga Workflow
+image::use-cases/orchestration-based-saga-pattern/order-fulfillment-saga-workflow.png[]
+
+In the workflow diagram, you can see how to describe the calls to the participants, the possible errors each service can throw, and how to define compensations for each step to be executed once an error happens during workflow execution.
```

Comment:
```suggestion
In the workflow diagram, you can see the calls to the participants, the possible errors each service can throw, and how to define compensations for each step to be executed once an error happens during workflow execution.
```
just not sure about the use of `calls` but maybe something @hmanwani-rh can review later on.

Reply:
+1